src/librustdoc/html/markdown.rs

```diff
@@ -442,7 +435,8 @@ impl<'a, I: Iterator<Item = Event<'a>>> Iterator for LinkReplacer<'a, I> {
             }
             // If this is a link, but not a shortcut link,
             // replace the URL, since the broken_link_callback was not called.
-            Some(Event::Start(Tag::Link(_, dest, title))) => {
+            Some(Event::Start(Tag::Link { dest_url: dest, title, .. })) => {
+                assert!(self.shortcut_link.is_none(), "links cannot be nested");
```

Comment:
Not completely sure about this one. Added it to make sure we're not silently mismatching link start/end link tags.